instrumentation/httpurlconnection/library/src/main/java/io/opentelemetry/instrumentation/library/httpurlconnection/HttpUrlInstrumentationConfig.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.library.httpurlconnection;
+
+import io.opentelemetry.instrumentation.api.incubator.semconv.net.PeerServiceResolver;
+import io.opentelemetry.instrumentation.api.internal.HttpConstants;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/** Configuration for automatic instrumentation of HttpURLConnection/HttpsURLConnection requests. */
+public final class HttpUrlInstrumentationConfig {
+    private static List<String> capturedRequestHeaders = new ArrayList<>();
+    private static List<String> capturedResponseHeaders = new ArrayList<>();
+    private static Set<String> knownMethods = HttpConstants.KNOWN_METHODS;
+    private static Map<String, String> peerServiceMapping = new HashMap<>();
+    private static boolean emitExperimentalHttpClientMetrics;
```

Comment:
Normally we would want configuration classes like this to be normal POJO or data style classes, rather than a pile of grouped globals (statics). We don't have to make that change in this PR, but ideally the config can be created and passed to the instrumentation that cares about it.

Since you're focused on build-time instrumentation, I can see how this came to be, though. For that case, a static block or other static initializer should create a single instance of the config from whatever the mechanism is (gradle properties, config file, whatever), and then that instance can be used by the instrumentation's static singleton stuff. Ensuring the order is also tricky. :)

Reply:
I made #418 to follow-up on this so we don't block this PR.